### ISSUE TITLE
TS IDE "exclude" folder does not work

### ISSUE THREAD

**Issue by nimo23** [angelozerr/typescript.java#104]
I made a test and generated the js-files manually by right-clicking on tsconfig.json "RunAs->Compile TypeScript". The js-files are generated. However, tsconfig.json has the property 

`"exclude": ["node_modules"]`

```
{
  "compilerOptions": {
    "target": "es5",
    "module": "commonjs",
    "moduleResolution": "node",
    "sourceMap": true,
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "removeComments": false,
    "noImplicitAny": false
  },
  	"compileOnSave": true,
	"buildOnSave": true,
	"exclude": ["node_modules"]
}

```

**But the console shows, that TS IDE generates js files for node_modules:**

```
Users/test/.p2/pool/plugins/ts.repository_1.1.0.201611012122/node_modules/typescript/lib/lib.d.ts
/Users/test/workspace/ide/angular-quickstart/node_modules/@angular/core/src/di/opaque_token.d.ts
/Users/test/workspace/ide/angular-quickstart/node_modules/@angular/core/src/type.d.ts
/Users/test/workspace/ide/angular-quickstart/node_modules/@angular/core/src/metadata/di.d.ts
/Users/test/workspace/ide/angular-quickstart/node_modules/@angular/core/src/animation/metadata.d.ts
/Users/test/workspace/ide/angular-quickstart/node_modules/@angular/core/src/change_detection/constants.d.ts
/Users/test/workspace/ide/angular-quickstart/node_modules/@angular/core/src/di/metadata.d.ts
/Users/test/workspace/ide/angular-quickstart/node_modules/@angular/core/src/di/forward_ref.d.ts
/Users/test/workspace/ide/angular-quickstart/node_modules/@angular/core/src/di/injector.d.ts
/Users/test/workspace/ide/angular-quickstart/node_modules/@angular/core/src/di/provider.d.ts
/Users/test/workspace/ide/angular-quickstart/node_modules/@angular/core/src/di/reflective_key.d.ts
/Users/test/workspace/ide/angular-quickstart/node_modules/@angular/core/src/di/reflective_provider.d.ts
/Users/test/workspace/ide/angular-quickstart/node_modules/@angular/core/src/di/reflective_injector.d.ts
/Users/test/workspace/ide/angular-quickstart/node_modules/@angular/core/src/di.d.ts
/Users/test/workspace/ide/angular-quickstart/node_modules/@angular/core/src/util/decorators.d.ts
/Users/test/workspace/ide/angular-quickstart/node_modules/@angular/core/src/metadata/view.d.ts
/Users/test/workspace/ide/angular-quickstart/node_modules/@angular/core/src/metadata/directives.d.ts
/Users/test/workspace/ide/angular-quickstart/node_modules/@angular/core/src/facade/lang.d.ts
/Users/test/workspace/ide/angular-quickstart/node_modules/@angular/core/src/change_detection/change_detection_util.d.ts
/Users/test/workspace/ide/angular-quickstart/node_modules/@angular/core/src/metadata/lifecycle_hooks.d.ts
/Users/test/workspace/ide/angular-quickstart/node_modules/@angular/core/src/metadata/ng_module.d.ts
/Users/test/workspace/ide/angular-quickstart/node_modules/@angular/core/src/metadata.d.ts
/Users/test/workspace/ide/angular-quickstart/node_modules/@angular/core/src/util.d.ts
/Users/test/workspace/ide/angular-quickstart/node_modules/@angular/core/src/error_handler.d.ts
/Users/test/workspace/ide/angular-quickstart/node_modules/@angular/core/src/application_init.d.ts
/Users/test/workspace/ide/angular-quickstart/node_modules/@angular/core/src/change_detection/change_detector_ref.d.ts
.......
```

**Comment by nimo23** [comment]
According to 

>No, it Print names of files part of the compilation. in other words --listFiles (see >http://www.typescriptlang.org/docs/handbook/compiler-options.html). I use this list files to >refresh the emited js files. I think the best option that I should use is --listEmittedFiles which is >now available with 2.0.6.

Yes, "--listEmittedFiles" would be better here. Is there a way where the user can set the compiler options of typescript by TS IDE? Would be nice to provide a "settings page" for the compiler options of tsc.

**Comment by angelozerr** [comment]
> Would be nice to provide a "settings page" for the compiler options of tsc.

You can set the compiler options with tsconfig.json editor. I don't want to provide a preferences page because your project can hosts several tsconfig.json files.

For --listEmittedFile, this option is enable since 2.0.0. So I must check the version of TypeScript to use it. I have started to develop something. Perhaps it will available this week in the 1.2.0-SNAPSHOT.

**Comment by nimo23** [comment]
>You can set the compiler options with tsconfig.json editor. 

This is sufficient. Thank you!
